fundingmanager.go

```diff
@@ -141,9 +140,18 @@ type fundingConfig struct {
 	// so that the channel creation process can be completed.
 	Notifier chainntnfs.ChainNotifier
 
+	// SignNodeKey is used to generate signature for node public key with
+	// funding keys.
+	SignNodeKey func(nodeKey, fundingKey *btcec.PublicKey) (*btcec.Signature, error)
```

Comment:
Rather than passing two new closure functions into the configuration (one for node keys and the other of bitcoin keys), the `fundingMgr` should only be passed a _single_ instance of `MessageSigner`. This implementation of `MessageSigner` should actually (behind the scenes) be composed of two concrete implementations: the one from the wallet, and a wrapper `MessageSigner` implementation which wraps the identity private key in the server. This dual implementation would then first attempt to sign with the wallet's `MessageSigner` falling back to the identity key signer if the wallet is unaware of the key.